src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -320,4 +363,47 @@ private function getTerminalWidth()
 
         return $dimensions[0] ?: self::MAX_LINE_LENGTH;
     }
+
+    private function autoPrependBlock()
+    {
+        $chars = substr($this->bufferedOutput->fetch(), -2);
+
+        if (false === $chars) {
+            return $this->newLine(); //empty history, so we should start with a new line.
+        }
+        //Prepend new line for each non PHP_EOL chars (This means no blank line was output before)
+        $this->newLine(2 - substr_count($chars, PHP_EOL));
+    }
+
+    private function autoPrependText()
+    {
+        $chars = $this->bufferedOutput->fetch();
+        //Prepend new line if last char isn't EOL:
+        if (PHP_EOL !== substr($chars, -1)) {
```

Comment:
Right. I should investigate this further.
I guess replacing every `PHP_EOL` by `\n` in the buffered output would be the most straightforward solution.